app/models/validations/date_validations.rb

```diff
@@ -42,18 +54,32 @@ def validate_startdate(record)
       record.errors.add :startdate, I18n.t("validations.lettings.date.startdate.after_major_repair_date")
     end
 
-    if record["voiddate"].present? && record["startdate"].to_date - record["voiddate"].to_date > 3650
-      record.errors.add :startdate, I18n.t("validations.lettings.date.startdate.ten_years_after_void_date")
-    end
-
-    if record["mrcdate"].present? && record["startdate"].to_date - record["mrcdate"].to_date > 3650
-      record.errors.add :startdate, I18n.t("validations.lettings.date.startdate.ten_years_after_mrc_date")
-    end
+    validate_startdate_against_mrc_and_void_dates(record)
   end
 
 private
 
   def is_rsnvac_first_let?(record)
     [15, 16, 17].include?(record["rsnvac"])
   end
+
+  def validate_startdate_against_mrc_and_void_dates(record)
+    if record.form.start_year_2025_or_later?
+      if record["voiddate"].present? && record["startdate"].to_date - record["voiddate"].to_date > 7300
+        record.errors.add :startdate, I18n.t("validations.lettings.date.startdate.twenty_years_after_void_date")
```

Comment:
Would it be worth refactoring these to do the check once and add the error to both fields?

Reply:
It would be!